firebase_admin/_utils.py

```diff
@@ -45,19 +79,50 @@ def _get_initialized_app(app):
         raise ValueError('Illegal app argument. Argument must be of type '
                          ' firebase_admin.App, but given "{0}".'.format(type(app)))
 
+
 def get_app_service(app, name, initializer):
     app = _get_initialized_app(app)
     return app._get_service(name, initializer) # pylint: disable=protected-access
 
-def handle_requests_error(error, message=None, status=None):
+
+def handle_platform_error_from_requests(error, handle_func=None):
+    """Constructs a ``FirebaseError`` from the given requests error.
+
+    This can be used to handle errors returned by Google Cloud Platform (GCP) APIs.
+
+    Args:
+        error: An error raised by the reqests module while making an HTTP call to a GCP API.
+        handle_func: A function that can be used to handle platform errors in a custom way. When
+            specified, this function will be called with four arguments -- parsed error response,
+            error message, source exception from requests and the HTTP response object.
+
+    Returns:
+        FirebaseError: A ``FirebaseError`` that can be raised to the user code.
+    """
+    if error.response is None:
+        return handle_requests_error(error)
+
+    response = error.response
+    content = response.content.decode()
+    status_code = response.status_code
+    error_dict, code, message = _parse_platform_error(content, status_code)
+    exc = None
+    if handle_func:
+        exc = handle_func(error_dict, message, error, error.response)
+
+    return exc if exc else handle_requests_error(error, message, code)
+
+
+def handle_requests_error(error, message=None, code=None):
     """Constructs a ``FirebaseError`` from the given requests error.
 
     Args:
         error: An error raised by the reqests module while making an HTTP call.
         message: A message to be included in the resulting ``FirebaseError`` (optional). If not
             specified the string representation of the ``error`` argument is used as the message.
-        status: An HTTP status code that will be used to determine the resulting error type
-            (optional). If not specified the HTTP status code on the error response is used.
+        code: An HTTP status code or GCP error code that will be used to determine the resulting
```

Comment:
Yeah, it still seems sketchy to me to mix types here. Would it be possible to make this one or the other?

It occurs to me that `handle_platform_error_from_requests` extracts `code`, which is then ignored if a `handle_func` is set. Maybe if `handle_requests_error` is a `handle_func`, then the code doesn't need to be passed in, and `handle_requests_error` could call `_parse_platform_error` itself?

I'm not sure if that's a good idea, but it's worth thinking about.

Reply:
`handle_requests_error` is supposed to be service-agnostic. I expect to use it for non-OP errors as well (we already do so in the `instance_id` module, which is a non-OP endpoint). So I extracted the OP-specific parsing to a new helper method named `_handle_func_requests` and made its signature consistent with `handle_func`.

But I do still need some branching since we allow `handle_func` to return `None`.